contracts/pools/PoolCollection.sol

```diff
@@ -1250,66 +1260,91 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
+        }
+
+        // if the target is equal to the current trading liquidity, no update is needed
+        if (targetBNTTradingLiquidity == liquidity.bntTradingLiquidity) {
+            return TargetTradingLiquidity({ update: false, bnt: 0, baseToken: 0 });
+        }
 
         // ensure that the target is above the minimum liquidity for trading
         if (targetBNTTradingLiquidity < minLiquidityForTrading) {
-            return TradingLiquidityAction({ update: true, newBNTTradingLiquidity: 0, newBaseTokenTradingLiquidity: 0 });
+            return TargetTradingLiquidity({ update: true, bnt: 0, baseToken: 0 });
         }
 
-        // calculate the new BNT trading liquidity and cap it by the growth factor
-        if (liquidity.bntTradingLiquidity == 0) {
-            // if the current BNT trading liquidity is 0, set it to the minimum liquidity for trading (with an
-            // additional buffer so that initial trades will be less likely to trigger disabling of trading)
-            uint256 newTargetBNTTradingLiquidity = minLiquidityForTrading * BOOTSTRAPPING_LIQUIDITY_BUFFER_FACTOR;
-
-            // ensure that we're not allocating more than the previously established limits
-            if (newTargetBNTTradingLiquidity > targetBNTTradingLiquidity) {
-                return
-                    TradingLiquidityAction({
-                        update: false,
-                        newBNTTradingLiquidity: 0,
-                        newBaseTokenTradingLiquidity: 0
-                    });
+        // calculate the target base token trading liquidity using the following:
+        // - calculate the delta between the current/target BNT trading liquidity
+        // - calculate the base token trading liquidity delta based on the BNT trading liquidity delta and the funding rate
+        // - apply the base token trading liquidity delta to the current base token trading liquidity
+        //
+        // note that the effective funding rate is always the rate between BNT and the base token)
+        uint256 bntTradingLiquidityDelta;
+        uint256 baseTokenTradingLiquidityDelta;
+
+        // liquidity increase
+        // note that liquidity increase is capped
+        if (targetBNTTradingLiquidity > liquidity.bntTradingLiquidity) {
+            if (liquidity.bntTradingLiquidity == 0) {
+                // the current BNT trading liquidity is 0 - set it to the minimum liquidity for trading (with an
+                // additional buffer so that initial trades will be less likely to trigger disabling of trading)
+                uint256 newTargetBNTTradingLiquidity = minLiquidityForTrading * BOOTSTRAPPING_LIQUIDITY_BUFFER_FACTOR;
+
+                // ensure that the new target isn't higher than the previously established target
+                targetBNTTradingLiquidity = Math.min(targetBNTTradingLiquidity, newTargetBNTTradingLiquidity);
```

Comment:
At least according to the comment, this statement should be:
```
newTargetBNTTradingLiquidity = Math.min(targetBNTTradingLiquidity, newTargetBNTTradingLiquidity);
```

Reply:
OK, I've confirmed that this confusion stems only from the name `newTargetBNTTradingLiquidity`.

I would just use a single assignment line in this 'if' clause:
```
targetBNTTradingLiquidity = Math.min(
    targetBNTTradingLiquidity,
    minLiquidityForTrading * BOOTSTRAPPING_LIQUIDITY_BUFFER_FACTOR
);
```

Reply:
Cleared up the docs/name

Reply:
Nice, updated.

Reply:
Or even merge the 'if' and the 'else' as follows:
```
targetBNTTradingLiquidity = Math.min(
    targetBNTTradingLiquidity,
    minLiquidityForTrading * (
        liquidity.bntTradingLiquidity == 0
        ? BOOTSTRAPPING_LIQUIDITY_BUFFER_FACTOR
        : LIQUIDITY_GROWTH_FACTOR
    )
);
```

Reply:
That I think will make it less readable, I prefer to keep the two cases separate

Reply:
Actually making another change there

Reply:
Ok, take a look

Reply:
Looks good!